### PR TITLE
Add audience for multi-cloud support

### PR DESCRIPTION
**Pull request by srnagar**
This PR adds builder methods to configure audience for multi-cloud support. This pattern is similar to the one used in ACR.

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/AzureLogAnalyticsImplBuilder.java [hunk 1]
```diff
@@ -185,22 +190,35 @@ public AzureLogAnalyticsImplBuilder addPolicy(HttpPipelinePolicy customPolicy) {
...
+     * Sets the audience to use for authentication with Azure Active Directory. The Azure Public Cloud audience will be
+     * used if the property is null.
+     * @param audience audience to use for authentication with Azure Active Directory. The Azure Public Cloud audience
+     * will be used if the property is null.
+     * @return the {@link AzureLogAnalyticsImplBuilder}.
+     */
+    public AzureLogAnalyticsImplBuilder audience(String audience) {
+        this.audience = audience;
+        return this;
+    }
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/AzureLogAnalyticsImplBuilder.java:203]: Does the autorest supports this directly? Or was this a manual edit?

Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/AzureLogAnalyticsImplBuilder.java:203]: This was a manual edit. AutoRest doesn't support this yet but will soon be supported.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryClientAudience.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import com.azure.core.util.ExpandableStringEnum;
+import com.azure.monitor.query.LogsQueryClient;
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Collection;
+
+/**
+ * Cloud audiences available for {@link LogsQueryClient}.
+ */
+public final class LogsQueryClientAudience extends ExpandableStringEnum<LogsQueryClientAudience> {
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryClientAudience.java:15]: Why do we have separate types for logs and metrics?

Comment (pallavit) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryClientAudience.java:15]: Also, do you need to set up a pipeline for the non public cloud. That can be done via a small change to the yaml file.

Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryClientAudience.java:15]: Logs and metrics use two different services and there are 2 separate clients. So, the audience for each can be different.

**Review by pallavit — APPROVED** [review]
:shipit:

**Pull request merged**